### ISSUE TITLE
Slack integration

### ISSUE THREAD

**Issue by seeker89** [bloomberg/powerfulseal#218]
It would be really cool to get PowerfulSeal to notify about errors on Slack.

**Comment by dgzlopes** [comment]
Love the idea :)

But I think that having some kind of AbstractNotification interface might be useful here (similar to the [metrics one](https://github.com/bloomberg/powerfulseal/blob/master/METRICS.md)). 

Useful in case someone wants to receive notifications in Mattermost, Zulip, Rocket, Discord...

**Comment by seeker89** [comment]
> Love the idea :)
> 
> But I think that having some kind of AbstractNotification interface might be useful here (similar to the [metrics one](https://github.com/bloomberg/powerfulseal/blob/master/METRICS.md)).
> 
> In case someone wants to receive notifications in Mattermost, Zulip, Rocket, Discord...

Definitely. Slack is all the rage now, but we'll see what's cool in a year or so :)

**Comment by simskij** [comment]
[Apprise](https://github.com/caronc/apprise) would likely be a good candidate for this.

**Comment by seeker89** [comment]
Apprise looks awesome, just like what we'd want here. Thanks! @simskij

**Comment by seeker89** [comment]
And it could make for a fun PR for Hacktoberfest too.

**Comment by simskij** [comment]
For sure, shouldn't be too much work to get it going.
